core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/common.kt

```diff
@@ -1,28 +1,46 @@
 package org.jetbrains.kotlinx.dataframe.io
 
-import com.github.kittinunf.fuel.httpGet
+import org.apache.commons.io.input.BOMInputStream
 import org.jetbrains.kotlinx.dataframe.AnyFrame
 import org.jetbrains.kotlinx.dataframe.DataFrame
 import org.jetbrains.kotlinx.dataframe.api.toDataFrame
 import org.jetbrains.kotlinx.dataframe.impl.columns.createColumnGuessingType
+import org.jetbrains.kotlinx.dataframe.util.IS_URL
+import org.jetbrains.kotlinx.dataframe.util.IS_URL_IMPORT
+import org.jetbrains.kotlinx.dataframe.util.IS_URL_REPLACE
 import java.io.File
-import java.io.IOException
 import java.io.InputStream
+import java.net.HttpURLConnection
 import java.net.URL
 
-internal fun catchHttpResponse(url: URL, body: (InputStream) -> AnyFrame): AnyFrame {
+/**
+ * Opens a stream to [url] to create a [DataFrame] from it.
+ * If the URL is a file URL, the file is read directly.
+ * If the URL is an HTTP URL, it's also read directly, but if the server returns an error code,
+ * the error response is read as JSON and parsed as [DataFrame] too.
+ *
+ * Public so it may be used in other modules.
+ */
+public fun catchHttpResponse(url: URL, body: (InputStream) -> AnyFrame): AnyFrame {
```

Comment:
Did you consider other ways to expose this API. Is it somehow possible to keep in a separate artifact so it's only available with explicit dependency?

Reply:
friend modules would work, but I'm not entirely a fan of those. A separate module could also work maybe. I could think of some more functions/classes that would benefit from being in a separate internal module (code generation, the double parser, these file/url/path/string helper functions, etc.) But that might be better solved in another issue.